### DISCUSSION TITLE
Even newly created SWC plugins cause `Segmentation fault` run by Next.js

### DISCUSSION THREAD

**Discussion by rinarakaki** [swc-project/swc#5574] [Q&A] [answered]
I'm getting the segmentation fault by the following steps. How can I solve this?

```sh
Segmentation fault
```

---

1) Create a new plugin

```sh
swc plugin new --target-type wasm32-wasi test
```

2) Modify the field `"main"` of the newly created plugin's `packages.json`

```json
{
  "main": "test.wasm",
}
```

3) Build it

```sh
cargo build-wasi --release
```

4) Copy output file from `/target/..` to the path specified by `"main"`

```sh
cp path/to/test.wasm path/to/main
```

5) Install the newly created plugin as an npm package

```sh
npm install --save-dev ./path/to/plugin
```

6) Configure `next.config.js`

```js
export default {
  experimental: {
    swcPlugins: [
      ['test', {}]
    ]
  }
}
```

7) Run

```sh
npm run dev
```

**Comment by kdy1** [comment]
Did you try `next@canary`?

**Comment by rinarakaki** [comment]
Thank you for your suggestion, and here is the output after I did `npm install next@canary`:

```sh
Caused by:
    Plugin runner cannot detect plugin's schema version. Ensure host is compiled with proper versions', 
...
error - ./node_modules/next/dist/client/dev/amp-dev.js
Error: failed to process failed to invoke plugin: failed to invoke plugin on 'Some("/path/to/node_modules/next/dist/client/dev/amp-dev.js")'
```

**Comment by kwonoj** [comment, marked as answer]
We will update next-swc to keep up with recent changes of swc/core soon, expect to have it in nex.js canary release.

**Reply by rinarakaki** [reply]
Thank you for your answer. Could you possibly point out for me the relevant pull request to be merged? I have kept getting the same error after 4 canary releases since I posted this question.
